### PR TITLE
Resolved IDENTITY-3102

### PR DESCRIPTION
**Pull request by IndunilRathnayake**

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 1]
```diff
@@ -1269,7 +1272,24 @@ public void addIdP(IdentityProvider identityProvider, int tenantId)
             }
 
             prepStmt.setString(4, CharacterEncoder.getSafeText(identityProvider.getHomeRealmId()));
-            setBlobValue(identityProvider.getCertificate(), prepStmt, 5);
+            CertData certData = null;
```
Comment (isharak) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: No need to assign null

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 2]
```diff
@@ -1269,7 +1272,24 @@ public void addIdP(IdentityProvider identityProvider, int tenantId)
...
+            if (StringUtils.isNotBlank(identityProvider.getCertificate())) {
+                try{
+                certData = IdentityApplicationManagementUtil.getCertData(identityProvider.getCertificate());
+                }catch (CertificateException ex){
+                    String msg = "Malformed Public Certificate file has been provided.";
+                    throw new IdentityApplicationManagementException(msg);
```
Comment (isharak) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: If this is not the service layer wrap the original message an throw it

File: components/tools/samlsso-validator/org.wso2.carbon.identity.tools.saml.validator/src/main/resources/META-INF/services.xml [hunk 3]
```diff
@@ -19,7 +19,7 @@
 		</description>
 		<parameter name="ServiceClass">org.wso2.carbon.identity.tools.saml.validator.SAMLValidatorService
 		</parameter>
-		<parameter name="DoAuthentication" locked="true">false</parameter>
+	        <parameter name="DoAuthentication" locked="true">false</parameter>
```
Comment (johannnallathamby) [inline components/tools/samlsso-validator/org.wso2.carbon.identity.tools.saml.validator/src/main/resources/META-INF/services.xml]: Why is there change in this file ? Can we please remove the change.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 4]
```diff
@@ -1269,7 +1272,17 @@ public void addIdP(IdentityProvider identityProvider, int tenantId)
             }
 
             prepStmt.setString(4, CharacterEncoder.getSafeText(identityProvider.getHomeRealmId()));
+
+            if (StringUtils.isNotBlank(identityProvider.getCertificate())) {
+                try {
+                    IdentityApplicationManagementUtil.getCertData(identityProvider.getCertificate());
+                } catch (CertificateException ex) {
+                    String msg = "Malformed Public Certificate file has been provided.";
```
Comment (hpmtissera) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: no need separate variable here (msg)

**Comment by IndunilRathnayake** [comment]
Above commented issues are resolved in https://github.com/wso2/carbon-identity/pull/950

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java (hunk 4 above)
Comment (IndunilRathnayake) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: This is fixed in #950

**Pull request merged**